gazelle_cabal/dependency_resolution.go

```diff
@@ -224,42 +223,85 @@ func getPackageLabel(
 	pkgName string,
 	from label.Label,
 ) label.Label {
-	parentPackage, packageName := resolvePackageName(pkgName)
+	var resolvers []func(*resolve.RuleIndex, string, string, label.Label)(label.Label, error)
+	resolvers = append(resolvers,
+		resolveAgainstPrivateLibraries,
+		resolveAgainstPublicLibraries,
+		resolveAgainstRepository,
+	)
+
+	for _, resolver := range resolvers {
+		lbl, err := resolver(ix, packageRepo, pkgName, from)
+		if (err == nil) {
+			return lbl
+		}
+	}
```

Comment:
How about unrolling the loop and calling the three functions explicitly?

There is even a fitting `goto` statement in go if you don't want to guard the calls with `if` statements.
https://golang.org/ref/spec#Goto_statements